prescient/simulator/config.py

```diff
@@ -74,334 +413,93 @@ def outer_parse(args=None, values=None):
-                    "Default is aCHP.",
-    )).declare_as_argument()
-
-    # # PRESCIENT_OUTPUT_OPTIONS
-
-    simulator_config.declare("output_ruc_initial_conditions", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output ruc (deterministic or stochastic) initial conditions prior "
-                    "to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_ruc_solutions", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output ruc solutions following each solve."
-                    " Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_sced_initial_conditions", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output sced initial conditions prior to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_sced_demands", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output sced demands prior to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_solver_logs", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output solver logs during execution.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_max_decimal_places", ConfigValue(
-        domain=PositiveInt,
-        default=6,
-        description="When writing summary files, this rounds the output to the "
-                    "specified accuracy. Default is 6.",
-    )).declare_as_argument()
-
-    simulator_config.declare("disable_stackgraphs", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Disable stackgraph generation",
-    )).declare_as_argument()
-
-    return simulator_config
-
-PrescientConfig = create_Prescient_Config()
-
-def clear_prescient_config():
-    global PrescientConfig
-    PrescientConfig = create_Prescient_Config()
+class _PluginPath(Path):
+    ''' A Path that registers a plugin when its path is set
+    '''
+    def __init__(self, parent_config: PrescientConfig):
+        self.config = parent_config
+        super().__init__()
+
+    def __call__(self, data):
+        path = super().__call__(data)
+        self.config.plugin_context.register_plugin(path, self.config)
+        return path
+
+class _InEnumStr(InEnum):
+    ''' A bit more forgiving string to enum parser
+    '''
+    def __call__(self, value):
+        if isinstance(value, str):
+            value = value.replace('-', '_').upper()
+        return super().__call__(value)
+
+def _StartDate(data):
+    ''' A basic start date validator/converter
+    '''
+    if isinstance(data, date):
+        return data
+    elif isinstance(data, datetime):
+        if data.hour != 0 or data.minute != 0:
+            print(f"WARNING: Prescient simulations always begin a midnight; ignoring time {data.time()}")
```

Comment:
Should this warning also be emitted if the incoming date was parse from a string?  Maybe switch the order here to:
```python
    if isinstance(data, date):
        return data
    if isinstance(data, str):
        try:
            data = dateutil.parser.parse(data)
        except ValueError:
            print(f"***ERROR: Illegally formatted start date={data} supplied!")
            raise
    assert isinstance(data, datetime):
    if data.hour != 0 or data.minute != 0:
        print(f"WARNING: Prescient simulations always begin a midnight; ignoring time {data.time()}")
    return data.date()

Reply:
Good catch; updated.